### ISSUE TITLE
Add support for buffers in uploadFile

### ISSUE THREAD

**Issue by abdelkd** [google-gemini/generative-ai-js#294]
### Description of the feature request:

Changing the `uploadFile` function so it accepts a `string` or `Buffer` so that `uploadingFile` can work without needing the file to be locally available.

### What problem are you trying to solve with this feature?

In the current version, the file has to be locally available in order to upload which is fine for most cases but not if the folder is read-only.

### Any other information you'd like to share?

I have already implemented the changes and waiting for approval to file a PR.

**Comment by YourAverageTechBro** [comment]
+1 to this issue. Makes Gemini hard to use in server instances when there is not a local file system available. Is there any work around for this?

**Comment by abdelkd** [comment]
@YourAverageTechBro I've encountered this issue when deploying to vercel, the workaround is patching the package using your package manager. I've done it using Pnpm in this [demo repo](https://github.com/abdelkd/genai-example)

**Comment by Rinlama** [comment]
Yes, I had same issues, so I used memoryStorage. I used the multer library with the memoryStorage option, which retains the files in memory instead of saving them to disk. The stored file data is subsequently converted into a Base64-encoded string for further processing.

**Comment by BrianHung** [comment]
Duplicate of #221. Let's consolidate there.

**Comment by hkt74** [comment]
Thanks @BrianHung, closed as duplicate
